sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -96,6 +96,7 @@ private[sql] object JDBCRDD extends Logging {
       case java.sql.Types.TINYINT       => IntegerType
       case java.sql.Types.VARBINARY     => BinaryType
       case java.sql.Types.VARCHAR       => StringType
+      case java.sql.Types.CHAR          => StringType
```

Comment:
case java.sql.Types.CHAR          => StringType
Repeat code between line99 and line 65, plz delete this code.